### PR TITLE
Feat/gguf tiled head order

### PR DESCRIPTION
**Pull request by co-esquie**
TLDR: `Qwen3.5-9B-IQ4_XS-GGUF` (and more broadly, all external GGUF but this is my test) VRAM usage dropped from 13gb to 7gb, aligned with llama cpp usage (6.5Gb)

---

Two independent fixes shipped together.

### GGUF tiled head order

External asymmetric Qwen3.5 GGUFs (llama.cpp convert) were loaded by dequantizing and unpermuting 8 SSM tensors per layer at init time. The fix keeps all tensors in their native quantized GGUF tiled order and instead branches the GQA expansion in `LinearAttn::forward` — `unsqueeze(2)` for tiled layout vs the existing `unsqueeze(3)` for HF grouped. Zero extra VRAM, projections stay quantized, single-file diff.

### `inferrs bench` timing correctness

The prefill throughput metric was meaningless: `prefill_ms` was captured without `device.synchronize()`, so it measured kernel-dispatch latency only, not actual GPU compute. Fixed along with three related issues: TTFT is now measured to first token (prefill + first sample), decode timer covers all N output tokens consistently, and `bypass_stop_tokens` ensures every run generates exactly `max_tokens` for stable comparisons.

### CODE REVIEW

File: inferrs/src/engine.rs [hunk 1]
```diff
@@ -2550,29 +2550,35 @@ impl Engine {
...
+        // Time all output tokens in decode_ms — including the first token sampled
+        // from prefill logits — to match llama.cpp's n_eval / n_p_eval convention.
+        let decode_start = Instant::now();
         let (mut token_id, _) = sampler::sample_token(&logits, sampling_params, &all_tokens)?;
         output_tokens.push(token_id);
         all_tokens.push(token_id);
+        let ttft_ms = prefill_start.elapsed().as_secs_f64() * 1000.0;
 
-        let decode_start = Instant::now();
         let mut finish_reason = self.check_stop(token_id, output_tokens.len(), sampling_params, "");
```
[the remark this answers is not in the document]
Comment (co-esquie) [reply inferrs/src/engine.rs]: unrelated to my changes but fixed

File: inferrs/src/engine.rs [hunk 2]
```diff
@@ -2585,6 +2591,7 @@ impl Engine {
             finish_reason = self.check_stop(token_id, output_tokens.len(), sampling_params, "");
```
[the remark this answers is not in the document]
Comment (co-esquie) [reply inferrs/src/engine.rs]: unrelated to my changes but fixed

**Comment by co-esquie** [comment]
`cargo run --release --features cuda --bin inferrs -- serve unsloth/Qwen3.5-9B-GGUF --gguf-file Qwen3.5-9B-IQ4_XS.gguf --device cuda`  used for testing
